providers/reddit/reddit.go

```diff
@@ -15,6 +16,14 @@ func init() {
 type Provider struct{}
 
 // BuildURI generates a search URL for Reddit.
+// If q starts with "/r/subreddit query", treat as subreddit and do restrict
+// search on the subreddit. The exception is when q = "/r/subreddit" where no
+// query is provided, treat it as regular reddit search.
 func (p *Provider) BuildURI(q string) string {
+	qs := strings.Split(q, " ")
+	if strings.HasPrefix(qs[0], "/r/") && len(qs) != 1 {
+		return fmt.Sprintf("https://www.reddit.com%s/search?q=%s&restrict_sr=on",
```

Comment:
I do not think that would work, because that would not restrict the search to the specific subreddit.
Try: https://www.reddit.com/r/cscareerquestions/search?q=best+startups